tests/test_search_optimizer.c

```
@@ -133,6 +139,43 @@ static int test_config_from_env_edge_values(void)
     return 0;
 }
 
+static int test_config_from_env_pso(void)
```

Clang (14.0.6)

[warning] function 'test_config_from_env_pso' has cognitive complexity of 72 (threshold 25)

(readability-function-cognitive-complexity)